src/ExchangeSharp/Model/ExchangeTicker.cs

```diff
@@ -21,109 +21,114 @@ MIT LICENSE
-        }
-
-        /// <summary>
-        /// Write to writer
-        /// </summary>
-        /// <param name="writer">Writer</param>
-        public void ToBinary(BinaryWriter writer)
-        {
-            writer.Write((double)Bid);
-            writer.Write((double)Ask);
-            writer.Write((double)Last);
-            Volume.ToBinary(writer);
-        }
-
-        /// <summary>
-        /// Read from reader
-        /// </summary>
-        /// <param name="reader">Reader</param>
-        public void FromBinary(BinaryReader reader)
-        {
-            Bid = (decimal)reader.ReadDouble();
-            Ask = (decimal)reader.ReadDouble();
-            Last = (decimal)reader.ReadDouble();
-            Volume = (Volume ?? new ExchangeVolume());
-            Volume.FromBinary(reader);
-        }
-    }
-
-    /// <summary>
-    /// Info about exchange volume
-    /// </summary>
-    public sealed class ExchangeVolume
-    {
-        /// <summary>
-        /// Last volume update timestamp
-        /// </summary>
-        public DateTime Timestamp { get; set; }
-
-        /// <summary>
-        /// Quote / Price currency - will equal base currency if exchange doesn't break it out by price unit and quantity unit
-        /// In BTC-USD, this would be USD
-        /// </summary>
-        public string QuoteCurrency { get; set; }
-
-        /// <summary>
-        /// Amount in units of the QuoteCurrency - will equal BaseCurrencyVolume if exchange doesn't break it out by price unit and quantity unit
-        /// In BTC-USD, this would be USD volume
-        /// </summary>
-        public decimal QuoteCurrencyVolume { get; set; }
-
-        /// <summary>
-        /// Base currency
-        /// In BTC-USD, this would be BTC
-        /// </summary>
-        public string BaseCurrency { get; set; }
-
-        /// <summary>
-        /// Base currency amount (this many units total)
-        /// In BTC-USD this would be BTC volume
-        /// </summary>
-        public decimal BaseCurrencyVolume { get; set; }
+	/// <summary>
+	/// Details of the current price of an exchange asset
+	/// </summary>
+	public sealed class ExchangeTicker
+	{
+		/// <summary>
+		/// An exchange specific id if known, otherwise null
+		/// </summary>
+		public string Id { get; set; }
+
+		/// <summary>
+		/// The currency pair symbol that this ticker is in reference to
+		/// </summary>
+		public string MarketSymbol { get; set; }
+
+		/// <summary>
+		/// The bid is the price to sell at
+		/// </summary>
+		public decimal Bid { get; set; }
+
+		/// <summary>
+		/// The ask is the price to buy at
+		/// </summary>
+		public decimal Ask { get; set; }
+
+		/// <summary>
+		/// The last trade purchase price
+		/// </summary>
+		public decimal Last { get; set; }
+
+		/// <summary>
+		/// This property contains the content of the complete api request results. This may contain additional information
+		/// </summary>
+		public string ApiResultString { get; set; }
```

Comment:
Instead of storing this as a large string, can you just store the JToken? That way, users don't have to incur a large `ToStringInvariant()` for every ticker response.

Reply:
I think you're right. I've updated the property name and type.